tests/integration/obs_preprocess/test_fetch_tropomi.py

```diff
@@ -57,3 +57,24 @@ def test_fetch_missing_creds(monkeypatch, env_var):
         match="EARTHDATA_USERNAME or EARTHDATA_PASSWORD environment variables missing",
     ):
         fetch_tropomi.create_session()
+
+
+# This hits the api with invalid data
+def test_fetch_invalid_date(tmpdir, root_dir):
```

Comment:
This test doesn't really test the retry behaviour, but I'll make an issue to better handle the testing of this module

Reply:
#90